code/modules/antagonists/heretic/items/hunter_rifle.dm

```diff
@@ -0,0 +1,146 @@
+	// In case someone puts this in turrets or something wacky, just fire like normal
+	if(!iscarbon(user))
+		return TRUE
+
+	if(currently_aiming)
+		user.balloon_alert(user, "already aiming!")
+		return FALSE
+
+	var/distance = get_dist(user, target)
+	var/fire_time = min((distance SECONDS) * seconds_per_distance, 10 SECONDS)
+
+	if(distance <= min_distance || !isliving(target))
+		return TRUE
+
+	currently_aiming = TRUE
+	user.balloon_alert(user, "taking aim...")
+	user.playsound_local(get_turf(user), 'sound/weapons/gun/general/chunkyrack.ogg', 100, TRUE)
+
+	var/image/reticle = image(
+		icon = 'icons/mob/actions/actions_items.dmi',
+		icon_state = "sniper_zoom",
+		layer = ABOVE_MOB_LAYER,
+		loc = target,
+	)
+	reticle.alpha = 0
+
+	var/list/mob/viewers = viewers(target)
+	// The shooter might be out of view, but they should be included
+	viewers |= user
+
+	for(var/mob/viewer as anything in viewers)
+		viewer.client?.images |= reticle
+
+	// Animate the fade in
+	animate(reticle, fire_time * 0.5, alpha = 255)
+	animate(fire_time * 0.25, transform = turn(reticle.transform, 180))
+	animate(fire_time * 0.25, transform = turn(reticle.transform, 180))
+
+	. = do_after(user, fire_time, target, IGNORE_TARGET_LOC_CHANGE, extra_checks = CALLBACK(src, .proc/check_fire_callback, target, user))
+
+	currently_aiming = FALSE
+	for(var/mob/viewer as anything in viewers)
+		viewer.client?.images -= reticle
+
+	if(!.)
+		user.balloon_alert(user, "interrupted!")
+
+	return .
+
+/// Callback for the do_after within the check_fire proc to see if something will prevent us from firing while aiming
+/obj/item/ammo_casing/a762/lionhunter/proc/check_fire_callback(mob/living/target, mob/living/user)
+	if(!isturf(target.loc))
+		return FALSE
+
+	return TRUE
+
+/obj/item/ammo_casing/a762/lionhunter/ready_proj(atom/target, mob/living/user, quiet, zone_override, atom/fired_from)
+	if(!loaded_projectile)
+		return
+
+	var/distance = get_dist(user, target)
+	// If we're close range, or the target's not a living, OR for some reason a non-carbon is firing the gun
+	// The projectile is dry-fired. Make the projectile considerably weaker
+	if(distance <= min_distance || !isliving(target) || !iscarbon(user))
+		loaded_projectile.projectile_phasing = NONE
+		loaded_projectile.knockdown = 0 SECONDS
+		loaded_projectile.stutter *= 0.33
+		loaded_projectile.damage *= 0.75
+		loaded_projectile.stamina *= 0.5
+
+	// BUT, if we're at a decent range and the target's a living mob,
+	// the projectile's been channel fired. It has full effects and homes in.
+	else
+		loaded_projectile.homing = TRUE
+		loaded_projectile.homing_turn_speed = 80
+		loaded_projectile.set_homing_target(target)
+
+	return ..()
+
+/obj/projectile/bullet/a762/lionhunter
+	name = "hunter's 7.62 bullet"
+	// These stats are only applied if the weapon is fired fully aimed
+	// If fired without aiming or at someone too close, it will do much less
+	damage = 40
```

Comment:
damage also seems really low considering 10 second aiming time. It's probably gonna be fine if you lower the aiming time